topic_modeling/train_models/train_models.py

```diff
@@ -0,0 +1,141 @@
+    data_dir = os.path.join(
+        os.path.normpath(os.getcwd() + os.sep + os.pardir),
+        'scrape_training_data')
+else:
+    base_dir = os.path.join(
+            os.getcwd(), 'topic_modeling', 'train_models')
+    data_dir = os.path.join(
+        os.getcwd(), 'topic_modeling', 'scrape_training_data')
+
+# Put all text from articles into one file,
+# if not already
+if not os.path.exists("all_articles.txt"):
+
+    flist = glob.glob(os.path.join(data_dir, 'the_dartmouth', '*.txt'))
+
+    all_articles = []
+
+    for file in flist:
+
+        if os.stat(file).st_size != 0:
+
+            file_text = open(file)
+            text_1 = file_text.read()
+
+            all_articles.append(text_1)
+
+            file_text.close()
+
+    file = open('all_articles.txt', 'w')
+    for article in all_articles:
+        file.write("{}\n".format(article))
+    file.close()
+
+
+all_articles_original = [line.rstrip('\n') for line in open(
+                        'all_articles.txt')]
+
+porter = PorterStemmer()
+
+all_articles_preprocessed = []
+
+for doc in all_articles_original:
+    s = doc.lower()
+    s = re.sub(r'([^\s\w]|_)+', ' ', s)
+    s = re.sub(r'\s+', ' ', s)  # gets rid of tabs
+    s = re.sub(r'[^A-Za-z0-9]+', ' ', s)
+    s = re.sub(r"\[([A-Za-z0-9_]+)\]", r'', s)  # get rid of words in brackets
+
+    tokens = word_tokenize(s)
+    stemmed = [porter.stem(word) for word in tokens]
+    s = " ".join(stemmed)
+
+    all_articles_preprocessed.append(s)
+
+my_additional_stop_words = [
+    've', 'll', 'd', 'm', 'o', 're', 'y', 'said',
+    'like', 'crosstalk', 'inaudible']
+stop_words = text.ENGLISH_STOP_WORDS.union(my_additional_stop_words)
+
+n_samples = 2000
+n_features = 1000
+n_components = [10]  # number of topics
+# n_components = [int(sys.argv[1])]
+num_top_words = 20
+
+for n_component in n_components:
+
+    tf_vectorizer = CountVectorizer(max_df=0.95, min_df=2,
+                                    max_features=None,
+                                    stop_words=stop_words)
+    tf = tf_vectorizer.fit_transform(all_articles_preprocessed)
+
+    lda = LatentDirichletAllocation(n_topics=n_component, max_iter=20,
+                                    learning_method='online',
+                                    learning_offset=50.,
+                                    random_state=0).fit(tf)
+
+    tf_feature_names = tf_vectorizer.get_feature_names()
+
+    top_words_dir = os.path.join(base_dir, 'models_topics')
+    pickle_dir = os.path.join(base_dir, 'models_pickles')
+
+    if not os.path.exists(top_words_dir):
+        os.makedirs(top_words_dir)
+
+    if not os.path.exists(pickle_dir):
+        os.makedirs(pickle_dir)
+
+    print_top_words(
+        lda, tf_feature_names, num_top_words, n_component, top_words_dir)
+
+    pickle_filename = 'LDA_dartmouth_topic{0}.pkl'.format(n_component)
+    with open(os.path.join(pickle_dir, pickle_filename), 'wb') as file:
+        pickle.dump(lda, file)
+    file.close()
```

Comment:
`file` is a bad name, and no need for `.close` since you are using `with open`

Reply:
good call! fixed.